### PR TITLE
[CL Message Audit]: MsgCollectFees

### PR DESCRIPTION
**Pull request by czarcas7ic**
Clickup Checklist:
https://app.clickup.com/37420681/v/dc/13nzm9-19087/13nzm9-35247

## What is the purpose of the change

This PR includes changes made from the internal audit of MsgCollectFees.

## Brief Changelog

- exports `InitOrUpdatePosition` method in accum package (used in import genesis)
- adds `fee_accum_record` field to `PositionData`
  - added to fix bug where individual position's fee accums weren't getting imported or exported via genesis
  - expands init/export genesis tests to check for this new field
- drive by change that grabs largest position ID when using the localosmosis script
  - nextPositionId was not getting properly set without this change due to failed position messages skipping position Ids
- readme reformat
- readme additions specific to fee accum / collection
- adds `PrepareClaimableFees` test
- expands `TestQueryAndCollectFees`
- adds `IsPositionOwner` test

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by p0mvn** [comment]
@czarcas7ic hey, can you please refresh this PR?

**Comment by czarcas7ic** [comment]
@p0mvn updated!

File: osmoutils/accum/accum.go [hunk 1]
```diff
@@ -367,14 +369,15 @@ func (accum AccumulatorObject) ClaimRewards(positionName string) (sdk.Coins, sdk
 
 	// Return the integer coins to the user
 	// The remaining change is thrown away.
-	// This is acceptable because we round in favour of the protocol.
+	// This is acceptable because we round in favor of the protocol.
```
Comment (p0mvn) [inline osmoutils/accum/accum.go:372]: You don't like Canadian English huh? 😆

File: osmoutils/accum/accum_helpers.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ var (
 
 // Creates a new position or override an existing position
 // at accumulator's current value with a specific number of shares and unclaimed rewards
-func initOrUpdatePosition(accum AccumulatorObject, accumulatorValue sdk.DecCoins, index string, numShareUnits sdk.Dec, unclaimedRewards sdk.DecCoins, options *Options) {
+func InitOrUpdatePosition(accum AccumulatorObject, accumulatorValue sdk.DecCoins, index string, numShareUnits sdk.Dec, unclaimedRewards sdk.DecCoins, options *Options) {
```
Comment (p0mvn) [inline osmoutils/accum/accum_helpers.go:15]: I would like to suggest that we make a copy of this method in `genesis.go` in CL.

It is important that we keep this unexported. Breakages of this abstraction lead to misuse and nasty bugs that are difficult to find and debug

File: proto/osmosis/concentrated-liquidity/genesis.proto [hunk 3]
```diff
@@ -51,6 +51,8 @@ message PoolData {
 message PositionData {
   Position position = 1;
   uint64 lock_id = 2 [ (gogoproto.moretags) = "yaml:\"lock_id\"" ];
+  osmosis.accum.v1beta1.Record fee_accum_record = 3
+      [ (gogoproto.nullable) = false ];
```
Comment (p0mvn) [inline proto/osmosis/concentrated-liquidity/genesis.proto:55]: Are incentives going to be added separately?

File: tests/cl-genesis-positions/edit_localosmosis_genesis.go [hunk 4]
```diff
@@ -149,9 +150,17 @@ func EditLocalOsmosisGenesis(updatedCLGenesis *clgenesis.GenesisState, updatedBa
...
+	for time.Since(start) < 30*time.Second {
+		if err := WriteFile(filepath.Join(localOsmosisHomePath, "config", "genesis.json"), genesisJson); err == nil {
+			fmt.Println("Genesis file written successfully")
+			return
+		} else {
+			fmt.Printf("Error writing genesis file: %s\n", err.Error())
+			time.Sleep(1 * time.Second)
+		}
 	}
+	fmt.Println("Timed out after 30 seconds")
```
Comment (p0mvn) [inline tests/cl-genesis-positions/edit_localosmosis_genesis.go:163]: What is this solving? Should we panic here?

**Review by p0mvn — APPROVED** [review]
LGTM after the comment about the accum abstraction breakage is addressed

File: proto/osmosis/concentrated-liquidity/genesis.proto (hunk 3 above)
Comment (czarcas7ic) [reply proto/osmosis/concentrated-liquidity/genesis.proto:55]: Incentives are being addressed by Sishir

File: osmoutils/accum/accum_helpers.go (hunk 2 above)
Comment (czarcas7ic) [reply osmoutils/accum/accum_helpers.go]: Agreed, unexported this and added a func directly to genesis.go https://github.com/osmosis-labs/osmosis/pull/5024/commits/c7d6a52cd9f5fb578338141f8269ce014602d86b

**Pull request merged**